traitsui/qt4/tests/test_helper.py

```diff
@@ -0,0 +1,118 @@
+from traitsui.qt4.helper import wrap_text_with_elision
+from traitsui.qt4.font_trait import create_traitsfont
+
+
+lorem_ipsum = (
+    "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod "
+    "tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim "
+    "veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea "
+    "commodo consequat.\n"
+    "    \n"
+    "Duis aute irure dolor in reprehenderit in voluptate velit esse cillum "
+    "dolore eu fugiat nulla pariatur.\n"
+    "Excepteur sint occaecat cupidatat non proident, sunt in culpa qui "
+    "officia deserunt mollit anim id est laborum."
+)
+
+def get_expected_lines(text, width):
+    expected_lines = []
+    for paragraph in text.splitlines():
+        if not paragraph.strip():
+            expected_lines.append('')
+            continue
+        expected_lines += textwrap.wrap(paragraph, width)
+    return expected_lines
+
+
+@skip_if_not_qt4
+class TestWrapText(unittest.TestCase):
+
+    def test_wrap_text_basic(self):
+        font = create_traitsfont("Courier")
+        font_metrics = QtGui.QFontMetrics(font)
+
+        average_char_width = font_metrics.averageCharWidth()
+        line_spacing = font_metrics.lineSpacing()
+
+        width = 500 * average_char_width
+        height = 100 * line_spacing
+
+        lines = wrap_text_with_elision(lorem_ipsum, font, width, height)
+
+        self.assertEqual(lines, [line.rstrip() for line in lorem_ipsum.splitlines()])
+
+    def test_wrap_text_empty(self):
+        font = create_traitsfont("Courier")
+        font_metrics = QtGui.QFontMetrics(font)
+
+        average_char_width = font_metrics.averageCharWidth()
+        line_spacing = font_metrics.lineSpacing()
+
+        width = 500 * average_char_width
+        height = 100 * line_spacing
+
+        lines = wrap_text_with_elision("", font, width, height)
+
+        self.assertEqual(lines, [])
+
+    def test_wrap_text_narrow(self):
+        font = create_traitsfont("Courier")
+        font_metrics = QtGui.QFontMetrics(font)
+
+        average_char_width = font_metrics.averageCharWidth()
+        line_spacing = font_metrics.lineSpacing()
+
+        width = 20 * average_char_width
+        height = 100 * line_spacing
+
+        lines = wrap_text_with_elision(lorem_ipsum, font, width, height)
+
+        expected_lines = get_expected_lines(lorem_ipsum, 20)
+        self.assertEqual(lines, expected_lines)
+
+    def test_wrap_text_narrow_short(self):
+        font = create_traitsfont("Courier")
+        font_metrics = QtGui.QFontMetrics(font)
+
+        average_char_width = font_metrics.averageCharWidth()
+        line_spacing = font_metrics.lineSpacing()
+
+        width = 20 * average_char_width
+        height = 20 * line_spacing
+
+        lines = wrap_text_with_elision(lorem_ipsum, font, width, height)
+
+        expected_lines = get_expected_lines(lorem_ipsum, 20)[:20]
+        expected_lines[-1] = u'fugiat nulla pariat\u2026'
```

Comment:
Not necessary with `from __future__ import unicode_literals`

Reply:
Belt _and_ suspenders!